### PR TITLE
Fix panic on bad length for SVCB record

### PR DESCRIPTION
**Pull request by bluejekyll**
cherry-picked from 0.20

see: #1465

### CODE REVIEW

File: crates/proto/src/rr/rdata/svcb.rs [hunk 1]
```diff
@@ -1032,7 +1032,7 @@ pub fn read(decoder: &mut BinDecoder<'_>, rdata_length: Restrict<u16>) -> ProtoR
     let svc_priority = decoder.read_u16()?.unverified(/*any u16 is valid*/);
     let target_name = Name::read(decoder)?;
 
-    let mut remainder_len = rdata_length.map(|len| len as usize - (decoder.index() - start_index)).unverified(/*valid len*/);
+    let mut remainder_len = rdata_length.map(|len| len as usize).checked_sub(decoder.index() - start_index).map_err(|len| format!("Bad length for RDATA of SVCB: {}", len))?.unverified(/*valid len*/);
```
Comment (djc) [inline crates/proto/src/rr/rdata/svcb.rs:1035]: Do we not run rustfmt in CI? This line is clearly too long...

Comment (trinity-1686a) [reply crates/proto/src/rr/rdata/svcb.rs:1035]: rustfmt is run in CI as part of the test/compatibility. For some reason rustfmt has no problem with the length of this line (?)

Comment (bluejekyll) [reply crates/proto/src/rr/rdata/svcb.rs:1035]: hm... the only thing we have in the rustfmt config file is the edition. don't know why it allows that line.

Comment (trinity-1686a) [reply crates/proto/src/rr/rdata/svcb.rs:1035]: it seems to come form the inline comment inside the call to unverified, if I remove that, rustfmt format as it should.

**Review by djc — APPROVED** [review]

**Pull request merged**